src/resources/cooperative.ts

```diff
@@ -12,26 +12,52 @@ export const CooperativeResource: ResourceIF = {
   nameRequestType: NameRequestTypes.CP,
   steps: CoopStepsTemplate,
   filingData: {
-    filingTypeCode: FilingCodes.INCORPORATION_CP,
-    entityType: CorpTypeCd.COOP
+    entityType: CorpTypeCd.COOP,
+    filingTypeCode: FilingCodes.INCORPORATION_CP
   },
-  directors: {
-    countMinimum: 3
-  },
-  shareClasses: {
-    countMinimum: null // TBD
-  },
-  incorporationAgreement: {
-    helpSection: [],
-    article: 'Sample_articles.pdf',
-    documents: [
+  peopleAndRoles: {
+    header: `1. Add People to your Application`,
+    blurb: `Add the people who will have a role in your Cooperative Association. A Completing
+      Party can also be a First Director.`,
+    helpSection: {
+      header: `Help with Adding First Directors`,
+      helpText: [
+        `To be in compliance with the Cooperative Association Act (Section 72);`,
+
+        `A minimum of thress directors is required. The first directors of the co-op must be
+          appointed in writing by a majority of the subscribers. The majority of Directors must
+          reside in Canada and at least one Director must reside in BC.`,
+
+        `Full names and residential addresses of each of the first directors must be included. The
+          residential address of a director must be a complete physical address. BC Registries and
+          Online Services cannot accept general delivery, post office box numbers, rural routes,
+          sites or comp. numbers as part of the address. You must also include a postal code. If
+          an area does not have stree names or numbers, provide a description that would readily
+          allow a person to locate the director.`
+      ]
+    },
+    addIncorporator: false,
+    addOrganization: false,
+    rules: [
+      {
+        rule: Rules.NUM_COMPLETING_PARTY,
+        text: 'The Completing Party',
+        test: (num) => { return (num === 1) }
+      },
+      {
+        rule: Rules.NUM_DIRECTORS,
+        text: 'At least three First Directors',
+        test: (num) => { return (num >= 3) }
+      },
       {
-        code: 'sample',
-        description: ''
+        rule: Rules.DIRECTOR_COUNTRY,
+        text: 'The majority of First Directors must reside in Canada',
+        test: (country) => { return (country === 'CA') }
```

Comment:
When the rule function was applied in the component, you may have felt that the rules was "half implemented" in the component and "half implemented" here...

I experimented with making the rule functions more generic -- eg, where you'd pass in the list of people, and the function would count the directors only and then determine whether the majority are in Canada, etc.

I also experimented with moving these arrow functions into a common file that all resource files (entities) could import.

However, I decided to keep this simple for now, since I don't know what other kinds of rules we might need. My experiments above just seemed to add a layer of complexity that just isn't needed at this time. (It may be needed in the future to simplify how the rules are applied in the components.

Thoughts?